frontend/src/metabase/admin/settings/components/CloudPanel/utils.ts

```diff
@@ -71,9 +71,11 @@ export const getMigrationEventTime = (isoString: string) =>
   dayjs(isoString).format("MMMM DD, YYYY, hh:mm A");
 
 export const getCheckoutUrl = (migration: CloudMigration) => {
-  const baseUrl = isProduction
-    ? `https://store.metabase.com`
-    : `https://store.staging.metabase.com`;
+  // Only used from components, but passing the setting value from all callers is a PITA.
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const baseUrl = useSetting("migration-use-staging")
```

Comment:
Instead of this, can we just have a setting that provides the store URL? That way we don't need to keep the FE and BE values in sync if they ever change

Reply:
Yeah, either name would probably be fine.

Reply:
Both are needed, because the BE uses `https://store-api.metabase.com/` and the FE uses `https://store.metabase.com/`. I added a single flag so we could control in a single way, since they need to agree.

Consider testing the JAR: with the single flag I have right now you'd do `MB_MIGRATION_USE_STAGING=true java -jar metabase.jar`.

But with url settings you'd do something like `MB_MIGRATION_BE_URL=https://store-api.metabase.com MB_MIGRATION_FE_URL=https://store.metabase.com java -jar metabase.jar`.

Thinking through it, I think adding the separate flags is worse. Maybe a setting just for the domain (`metabase.com` vs `staging.metabase.com`) would be better?

Reply:
Thought about it a bit more and I think setting the domain isn't great because it wouldn't allow actually changing the domain in some cases. For instance when developing locally, because you still need to know it's http vs https. And I know I need this for an upcoming project already so I can test store integration without deploying first.

I think I'll add the URL settings, but make their default value depend on another setting like `MB_STORE_USE_STAGING` for ease of jar testing.

Reply:
Done